lib/ui/settings/data/import/google_auth_image_import.dart

```diff
@@ -0,0 +1,133 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:base32/base32.dart';
+import 'package:ente_auth/l10n/l10n.dart';
+import 'package:ente_auth/models/code.dart';
+import 'package:ente_auth/models/protos/googleauth.pb.dart';
+import 'package:ente_auth/services/authenticator_service.dart';
+import 'package:ente_auth/store/code_store.dart';
+import 'package:ente_auth/ui/components/buttons/button_widget.dart';
+import 'package:ente_auth/ui/components/dialog_widget.dart';
+import 'package:ente_auth/ui/components/models/button_type.dart';
+import 'package:ente_auth/ui/settings/data/import/analyze_qr_code.dart';
+import 'package:ente_auth/ui/settings/data/import/import_success.dart';
+import 'package:flutter/material.dart';
+import 'package:logging/logging.dart';
+
+const kGoogleAuthExportPrefix = 'otpauth-migration://offline?data=';
+
+Future<void> showGoogleAuthImageInstruction(BuildContext context) async {
+  final l10n = context.l10n;
+  final result = await showDialogWidget(
+    context: context,
+    title: l10n.importFromApp("Google Authenticator"),
+    body: l10n.importGoogleAuthGuide,
+    buttons: [
+      ButtonWidget(
+        buttonType: ButtonType.primary,
+        labelText: l10n.scanAQrCode,
+        isInAlert: true,
+        buttonSize: ButtonSize.large,
+        buttonAction: ButtonAction.first,
+      ),
+      ButtonWidget(
+        buttonType: ButtonType.secondary,
+        labelText: context.l10n.cancel,
+        buttonSize: ButtonSize.large,
+        isInAlert: true,
+        buttonAction: ButtonAction.second,
+      ),
+    ],
+  );
+  if (result?.action != null && result!.action != ButtonAction.cancel) {
+    if (result.action == ButtonAction.first) {
+      final List<Code>? codes = await Navigator.of(context).push(
+        MaterialPageRoute(
+          builder: (BuildContext context) {
+            return const QrScanner();
+          },
+        ),
+      );
+      if (codes == null || codes.isEmpty) {
+        return;
+      }
+      for (final code in codes) {
+        await CodeStore.instance.addCode(code, shouldSync: false);
+      }
+      unawaited(AuthenticatorService.instance.onlineSync());
+      importSuccessDialog(context, codes.length);
+    }
+  }
+}
+
+List<Code> parseGoogleAuth(String qrCodeData) {
```

Comment:
We can reuse the logic from other Google Auth Import?

Reply:
Yes, but now do we need that. As in the new, we have a scanner and image analyzing together right, so we can remove the existing google authenticator.

Reply:
Yes, we will need that. Google Auth app doesn't let you take screenshot afaik, so we feel that other import will still be required.

Ideally, when user clicks on Import from Google Authenticator, we can show two options: `Scan QR Code`, `Scan code from image` and Cancel button in the end. Based on user's selection, we can trigger relevant flow.

Reply:
Yea, we will anyhow will have a scanner. I have added a feature for both scanning a Qr code directly and also by analyzing the Qr code image selected from the gallery.  That's why I told, do we need the earlier option, where we just scan a Qr code. Also it is using a package which is in maintenance mode. any time, that package can be removed.

Reply:
Understood. In this case, let's remove the older option and call this import option as  `Google Authenticator` and remove ` (saved image)` part.